src/adjoint/Makefile_tapenade

```diff
@@ -174,34 +174,37 @@ sa%saResScale(scratch, dw) > \
 residuals%sourceTerms_block(w, pref, uref, plocal, dw, actuatorRegions%force, actuatorRegions%heat) > \
                            (w, pref, uref, plocal, dw, actuatorRegions%force, actuatorRegions%heat) \
 \
-fluxes%inviscidCentralFlux(w, timeRef, vol, si, sj, sk, p, dw) > \
-                          (w, timeRef, vol, si, sj, sk, p, dw) \
+residuals%initres_block(dw, fw, flowDoms%w, flowDoms%vol, dscalar, dvector) > \
+                       (dw, fw, flowDoms%w, flowDoms%vol, dscalar, dvector) \
+\
+fluxes%inviscidCentralFlux(w, timeRef, vol, si, sj, sk, p, dw, sfacei, sfacej, sfacek) > \
+                          (w, timeRef, vol, si, sj, sk, p, dw, sfacei, sfacej, sfacek) \
 \
 fluxes%inviscidDissFluxScalar(w, pInfCorr, rhoInf, p, radi, radj, radk, fw) > \
-                             (w, pInfCOrr, rhoInf, p, fw) \
+                             (w, pInfCOrr, rhoInf, p, radi, radj, radk, fw) \
 \
-fluxes%inviscidDissFluxMatrix(w, pInfCorr, si, sj, sk, p, fw) > \
-                             (w, pInfCorr, si, sj, sk, p, fw) \
+fluxes%inviscidDissFluxMatrix(w, pInfCorr, si, sj, sk, p, fw, sfacei, sfacej, sfacek) > \
+                             (w, pInfCorr, si, sj, sk, p, fw, sfacei, sfacej, sfacek) \
 \
-fluxes%inviscidDissFluxScalarApprox(w, p, radi, radj, radk, fw) > \
+fluxes%inviscidDissFluxScalarApprox(w, p, radi, radj, radk, fw, rhoinf, pinfcorr) > \
                                    (w, p, fw) \
 \
-fluxes%inviscidDissFluxMatrixApprox(w, p, fw) > \
-                                   (w, p, fw) \
+fluxes%inviscidDissFluxMatrixApprox(w, p, fw, pinfcorr, sfacei, sfacej, sfacek, si, sj, sk) > \
+                                   (w, p, fw, pinfcorr, sfacei, sfacej, sfacek, si, sj, sk) \
 \
-fluxes%inviscidUpwindFlux(w, si, sj, sk, p, fw) > \
-                         (w, si, sj, sk, p, fw) \
+fluxes%inviscidUpwindFlux(w, si, sj, sk, p, fw, sfacei, sfacej, sfacek) > \
+                         (w, si, sj, sk, p, fw, sfacei, sfacej, sfacek) \
 \
-flowUtils%computeSpeedOfSoundSquared(w, p) > \
+flowUtils%computeSpeedOfSoundSquared(w, p, aa) > \
                                     (w, p, aa) \
 \
-flowUtils%allNodalGradients(w, vol, si, sj, sk, aa) > \
+flowUtils%allNodalGradients(w, vol, si, sj, sk, aa, ux, uy, uz, vx, vy, vz, wx, wy, wz, qx, qy, qz) > \
                            (w, vol, si, sj, sk, aa, ux, uy, uz, vx, vy, vz, wx, wy, wz, qx, qy, qz) \
 \
-fluxes%viscousFlux(w, x, si, sj, sk, rlv, rev, aa, ux, uy, uz, vx, vy, vz, wx, wy, wz, qx, qy, qz, fw) > \
-                  (w, x, si, sj, sk, fw, viscsubface%tau) \
+fluxes%viscousFlux(w, x, si, sj, sk, rlv, rev, aa, ux, uy, uz, vx, vy, vz, wx, wy, wz, qx, qy, qz, fw, viscsubface%tau, viscSubface%q) > \
+                  (w, x, si, sj, sk, rlv, rev, aa, ux, uy, uz, vx, vy, vz, wx, wy, wz, qx, qy, qz, fw, viscsubface%tau, viscSubface%q) \
```

Comment:
why do we need all of these additions here?

Reply:
@SichengHe prefers to include all inputs/outputs to avoid partials to be zeroed out by mistake as it was happening before. Maybe not the most efficient approach, but the safest. Should we be more selective here?

Reply:
@marcomangano That is correct. Not 100% sure about the behavior of adflow, maybe @eirikurj has more comments on this.

Reply:
I dont know the details here but it sounds like being more selective would be better in terms of development effort later on. It's similar to the use statements that add everything; it becomes difficult to understand what terms are relevant and what terms are not.

Reply:
Ok I gave a quick look at the code here, and it looks like all these terms are related to viscous flux calculations. I think only `flowUtils%allNodalGradients` and `fluxes%viscousFlux` have extra terms, all the other additions look targeted.
Figuring out which of these terms should not be differentiated goes a bit beyond my current understanding of the code, and I don't think that iteratively removing components until the gradient  breaks again is a sound approach.

@SichengHe @anilyil do you have a bit more insight/suggestions here?